Modify rule S2176(Java): Update rule according to the LayC (#2445)
SonarSource/rspec: rules/S2176/java/rule.adoc

== Why is this an issue?

Two classes can have the same simple name if they are in two different packages.

[source,java]
----
package org.foo.domain;

public class User {
  // ..
}
----

[source,java]
----
package org.foo.presentation;

public class User {
  // ..
}
----

> But it is an issue when a class has the same name as the superclass it extends or the interfaces it implements.

Reviewer: ```suggestion
This becomes an issue when a class has the same name as the superclass it extends or the interfaces it implements.
```

> This issue is called class name shadowing. It's an issue because it can be unclear which class is being referred to

Reviewer: AFAIK, one should avoid contractions in formal writing
```suggestion
This issue is called class name shadowing. It is an issue because it can be unclear which class is being referred to
```

> However, this is an issue when a class has the same name as the superclass it extends or the interfaces it implements.
> This issue is called class name shadowing. It is an issue because it can be unclear which class is being referred to

Reviewer: We use the word "issue" a lot here. My proposal:
```suggestion
However, this becomes an issue when a class has the same name as the superclass it extends or the interfaces it implements, also known as class name shadowing.
It is problematic because it can be unclear which class is being referred to
```
